src/liballoc/vec.rs

```diff
@@ -1306,6 +1306,48 @@ impl<T> Vec<T> {
         }
         other
     }
+
+    /// Resizes the `Vec` in-place so that `len` is equal to `new_len`.
+    ///
+    /// If `new_len` is greater than `len`, the `Vec` is extended by the
+    /// difference, with each additional slot filled with the result of
+    /// closure `f`. If `new_len` is less than `len`, the `Vec` is simply
+    /// truncated.
+    ///
+    /// This method uses a closure to create new values on every push. If
+    /// you'd rather [`Clone`] a given value, use [`resize`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(vec_resize_with)]
+    ///
+    /// let mut vec = vec![1, 2, 3];
+    /// vec.resize_with(5, Default::default);
+    /// assert_eq!(vec, [1, 2, 3, 0, 0]);
+    ///
+    /// let mut vec = vec![1, 2, 3, 4];
+    /// vec.resize_with(2, Default::default);
+    /// assert_eq!(vec, [1, 2]);
+    /// ```
```

Comment:
Since it's FnMut, consider adding an example that uses state of some kind.  Perhaps
```rust
let mut vec = vec![];
let mut p = 1;
vec.resize_with(4, || { p *= 2; p });
assert_eq!(vec, [2, 4, 8, 16]);
```

Reply:
Yup, that was on my list but forgot about it. I've updated it now, using your example.